upsilonconf/config.py

```diff
@@ -447,73 +449,83 @@ def to_dict(self, key_mods: Mapping[str, str] = None) -> Dict[str, Any]:
         if key_mods is None:
             key_mods = {}
 
-        return _replace_in_keys(self, key_mods)
+        return _modify_keys(self, key_mods)
 
 
 # utilities
 
 
-def __replace_in_keys(
-    mapping: Union[Mapping[str, Any], Any], s: str, r: str
-) -> Union[Mapping[str, Any], Any]:
+def _modify_keys(
+    mapping: Mapping[str, Any], key_mods: Mapping[str, str]
+) -> Dict[str, Any]:
     """
-    Take a mapping object and replace all occurrencies of `s` with `r` in any
-    of its keys.
-
-    This function is called recursively.
-
-    Modelled after the Stack Overflow answer by Farhan Haider:
-    https://stackoverflow.com/questions/21650850/pyyaml-replace-dash-in-keys-with-underscore#answer-55986782
+    Replace strings in the keys of a mapping object recursively.
 
     Parameters
     ----------
     mapping : Mapping
-        The mapping object to be modified.
-    s : str
-        The string to be replaced in the keys.
-    r : str
-        The replacement string.
+        The mapping object (`Configuration`, `dict`, ...) whose keys are to be
+        modified.
+    key_mods : Mapping
+        The dictionary with the replacements: All key strings are replaced
+        with the corresponding values from the this dictionary.
 
     Returns
     -------
-        The dictionary with all strings `s` replaced with `r` in the keys.
+    dict
+        A dictionary with the modified keys.
+
     """
-    # Traverse all the keys and replace `s` with `r`
-    dictionary = {}
-    for key, value in mapping.items():
-        try:
-            # Call this method recursively
-            value = __replace_in_keys(value, s, r)
-        except AttributeError:
-            # `value` is not of the mapping type
-            pass
-        dictionary[key.replace(s, r)] = value
+    # Build and compile replacement pattern
+    pattern = re.compile(
+        "|".join(
+            [re.escape(k) for k in sorted(key_mods, key=lambda k: len(k), reverse=True)]
```

Comment:
Maybe you don't like to add variables, but I would insert a `sorted_mods` variable here for readability.

Reply:
> Maybe you don't like to add variables

Not necessarily …

Done. https://github.com/hoedt/upsilonconf/pull/6/commits/ca684a4c47386b7143060c35f41e38cfe3a85ffd